reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/scope/DisposableScopeImpl.kt

```diff
@@ -0,0 +1,93 @@
+package com.badoo.reaktive.disposable.scope
+
+import com.badoo.reaktive.base.CompleteCallback
+import com.badoo.reaktive.completable.Completable
+import com.badoo.reaktive.completable.subscribe
+import com.badoo.reaktive.disposable.CompositeDisposable
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.maybe.Maybe
+import com.badoo.reaktive.maybe.subscribe
+import com.badoo.reaktive.observable.Observable
+import com.badoo.reaktive.observable.subscribe
+import com.badoo.reaktive.single.Single
+import com.badoo.reaktive.single.subscribe
+
+internal class DisposableScopeImpl : DisposableScope, CompositeDisposable() {
+
+    override fun <T : Disposable> T.scope(): T {
+        purge()
+        add(this)
+
+        return this
+    }
+
+    override fun <T : Disposable> T.unaryPlus(): T = scope()
+
+    override fun <T : CompleteCallback> T.scope(): T {
+        Disposable(::onComplete).scope()
+
+        return this
+    }
+
+    override fun <T : CompleteCallback> T.unaryPlus(): T = scope()
+
+    override fun <T> Observable<T>.subscribeScoped(
```

Comment:
Does Kotlin name resolution allow to name it `subscribe`?

Reply:
It does, but I thought it will be very confusing to users.

Reply:
Not sure, but probably you're right